mplc/partner.py

```diff
@@ -54,19 +54,7 @@ def shuffle_labels(self, proportion_shuffled):
             raise ValueError(
                 f"The proportion of labels to corrupted was {proportion_shuffled} but it must be between 0 and 1."
             )
-
         # Select the indices where the label will be shuffled
         n = int(len(self.y_train) * proportion_shuffled)
-        idx = sample(list(range(len(self.y_train))), n)
-
-        # Suffle the labels
-        for i in idx:
-            label = self.y_train[i]
-            # Suffle the label
-            new_label = shuffle(label)
-
-            # Force the label to be different
-            while np.all(new_label == label):
-                new_label = shuffle(label)
-
-            self.y_train[i] = new_label
+        for label in self.y_train[:n]:
```

Comment:
We want to shuffle only the first n labels ? 
I think it makes more sense to randomly choose n labels among len(y_train) and to shuffle them.

Reply:
I though labels were shuffled during the data splittin between partners, and before each minibatch split. Nevertheless, in advanced split, it is not obvious that the data are shuffled. I changed that